src/components/FeeLabelToggle/FeeLabelToggle.tsx

```diff
@@ -91,41 +93,69 @@ const FeeLabelToggle = ({
   const chainAccountAssets = accountAssets[chain] ?? {};
 
   const feeDisplayValue = formatTransactionFee(chain, txFeeInWei, gasToken);
-  const feeValue = getFormattedTransactionFeeValue(chain, txFeeInWei, gasToken);
   const currencySymbol = getCurrencySymbol(fiatCurrency);
 
-  const gasAddress = getGasAddress(chain, gasToken);
-  const feeInFiat = parseFloat(feeValue) * getAssetRateInFiat(chainRates, gasAddress, fiatCurrency);
+  const feeInFiat = getTxFeeInFiat(chain, txFeeInWei, gasToken, chainRates, fiatCurrency);
   const feeInFiatDisplayValue = `${currencySymbol}${feeInFiat.toFixed(2)}`;
   const labelValue = isFiatValueVisible ? feeInFiatDisplayValue : feeDisplayValue;
 
-  showRelayerMigration = showRelayerMigration &&
+  showRelayerMigration =
+    showRelayerMigration &&
     firebaseRemoteConfig.getBoolean(REMOTE_CONFIG.APP_FEES_PAID_WITH_PLR) &&
     !isGasTokenSupported;
 
   const openRelayerMigrationModal = () => {
     if (!showRelayerMigration) return;
-    Modal.open(() => (
-      <RelayerMigrationModal
-        accountAssets={chainAccountAssets}
-        accountHistory={accountHistory}
-      />
-    ));
+    Modal.open(() => <RelayerMigrationModal accountAssets={chainAccountAssets} accountHistory={accountHistory} />);
   };
 
-  return (
-    <LabelWrapper >
-      <Label>{labelText || t('label.estimatedFee')}&nbsp;</Label>
-      <Spacing w={8} />
-      <FeePill onPress={() => setIsFiatValueVisible(!isFiatValueVisible)} hasError={hasError}>
-        <BaseText small color="#ffffff">{labelValue}</BaseText>
-      </FeePill>
-      <Spacing w={8} />
-      {showRelayerMigration && (
+  const highFee = isHighGasFee(chain, txFeeInWei, gasToken, chainRates, fiatCurrency, gasThresholds);
+
+  const onHighFeeInfoPress = () => {
+    if (!highGasFeeModal) return;
+
+    Modal.open(() => highGasFeeModal);
+  };
+
+  const renderRelayerMigration = () => {
+    return (
+      <>
+        <Spacing w={spacing.small} />
         <BaseText small link onPress={openRelayerMigrationModal}>
           {t('label.payWithPLR')} <Emoji name="ok_hand" style={{ fontSize: 12 }} />
         </BaseText>
-      )}
+      </>
+    );
+  };
+
+  const renderHighGasFeeWarning = () => {
+    return (
+      <>
+        <Label color={highFee && colors.negative}>{labelText || t('label.highGasFee')}&nbsp;</Label>
+        {highGasFeeModal && (
+          <HighGasFeeButton onPress={onHighFeeInfoPress} padding={spacing.small}>
+            <Icon name={'info'} color={colors.primary} />
+          </HighGasFeeButton>
+        )}
+      </>
+    );
+  };
+
+  return (
+    <LabelWrapper>
+      <Label color={highFee && colors.negative}>{labelText || t('label.estimatedFee')}&nbsp;</Label>
+
+      <Spacing w={spacing.small} />
+
+      <FeePill onPress={() => setIsFiatValueVisible(!isFiatValueVisible)} hasError={hasError} highFee={highFee}>
+        <BaseText small color="#ffffff">
```

Comment:
Yeah can do. That was something that was there beforehand - I'm assuming because `FeePill` always has a dark background colour, so light/dark modes shouldn't hopefully affect it.